### DISCUSSION TITLE
Use Pester to mock kubectl command

### DISCUSSION THREAD

**Discussion by ktran1005** [pester/Pester#2367] [Q&A]
### Checklist

- [X] Feature request has a meaningful title
- [X] I have searched the existing issues. See [all issues](https://github.com/pester/Pester/issues?q=is%3Aissue)
- [X] I have tested using the latest version of Pester. See [Installation and update guide](https://pester.dev/docs/introduction/installation).

### Summary of the feature request

I am trying to use Pester to mock kubectl command for one of my ps script. I wonder that is it doable in this case? One of the command is `kubectl get pods -n ...`. If anyone has some experience with this then it would be great.

**Comment by fflaten** [comment]
Hi. Mocking depends on how your script consumes the command and the output. Do you have any details?
There's an example for mocking native applications in general here: https://pester.dev/docs/usage/mocking#mocking-a-native-application

**Reply by ktran1005** [reply]
Hi @fflaten, in my case I want to mock the command `kubectl get pods -n $namespace`. I have tried the way we mock with azure command but it does not work. My mock looks like this 
`Mock kubectl { cmd /c exit 0 } -ParameterFilter {"$args" -match 'get pods'}` or even I removed the Parameters, it still calls the actual command

**Reply by fflaten** [reply]
Is it called from a function in a module?

**Reply by ktran1005** [reply]
@fflaten the powerscript logs into azure and use kubectl to create stuffs. Therefore, it is called in powershell script. Not sure if I answer your question correctly.

**Reply by fflaten** [reply]
Yeah, but it should work for a script. Are you able to create a minimal repro test + script?

**Comment by nohwnd** [comment]
Closing, we never got a repro. Feel free to reopen or start a new discussion with a minimal test and script if you still hit this.
